### ISSUE TITLE
Destructured declarations in for-of loop fail to parse

### ISSUE THREAD

**Issue by AgentME** [facebook/flow#805]
```javascript
/* @flow */
for (var [x, y] of [[5,6],[7,8]]) {
  console.log(x, y);
}
```

```
$ flow

/private/tmp/a/test.js:2:10,15: Destructuring assignment must be initialized
```

This error even happens for files without a `@flow` comment if they're imported from any files that do. Moving the `var` keyword out of the for-loop works around the issue for files without `@flow` comments:

```javascript
var x, y;
for ([x, y] of [[5,6],[7,8]]) {
  console.log(x, y);
}
```

**Comment by gabelevi** [comment]
Hmm, you're right, it looks like our tests didn't cover destructuring in for-of and missed the regression in https://github.com/facebook/flow/pull/735

We'll fix this for the next release, sorry for the trouble!

**Comment by samwgoldman** [comment]
@gabelevi I actually have a fix for this, which I tied into a fix for #830. Hopefully you didn't spend a lot of time on this already (assuming my fix is good).
